sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -313,7 +314,7 @@ public TableRow apply(T input) {
   public static Row toBeamRow(GenericRecord record, Schema schema, ConversionOptions options) {
     List<Object> valuesInOrder =
         schema.getFields().stream()
-            .map(field -> convertAvroFormat(field, record.get(field.getName()), options))
+            .map(field -> convertAvroFormat(field.getType(), record.get(field.getName()), options))
```

Comment:
Here is where you can give an error message with the field name.

```java
  .map(field -> {
     try {
       return convertAvroFormat(field.getType(), record.get(field.getName()), options))
     } catch (Exception cause) {
       throw new IllegalArgumentException("Error converting field " + field + ": " cause.getMessage(), cause)
     }
   })
```

Reply:
ah that makes sense. thank you I added that as well!